core/server/api/v2/utils/index.js

```diff
@@ -11,18 +11,8 @@ module.exports = {
         return require('./validators');
     },
 
-    /**
-     *  TODO: We need to check for public context as permission stage overrides
-     * the whole context object currently: https://github.com/TryGhost/Ghost/issues/10099
-     */
     isContentAPI: (frame) => {
-        const context = frame.options && frame.options.context || {};
-        // CASE: An empty context is considered public by the core/server/services/permissions/parse-context.js
-        //       module, replicated here because the context is unparsed until after the permissions layer of the pipeline
-        const isPublic = context.public || Object.keys(context).length === 0;
-
-        // CASE: apiType = 'content' for HTTP Content API
-        return frame.apiType === 'content' || isPublic;
+        return frame.apiType === 'content';
```

Comment:
Doesn't matter which context you have. If you request content API, you are in content ctrl land. Same for admin.